fiftyone/core/dataset.py

```
@@ -1219,43 +1219,26 @@
 
         Returns:
             a :class:`fiftyone.core.sample.Sample`
         """
-        cursor = self._get_first(reverse=True)
-
         try:
-            d = next(cursor)
-        except StopIteration:
+            sample_view = self[-1:].first()
+        except ValueError:
             raise ValueError("%s is empty" % self.__class__.__name__)
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)